libs/langchain/langchain/retrievers/you.py

```diff
@@ -34,13 +35,20 @@ def _get_relevant_documents(
         import requests
 
         headers = {"X-API-Key": self.ydc_api_key}
-        results = requests.get(
-            f"https://api.ydc-index.io/search?query={query}",
-            headers=headers,
-        ).json()
-
-        docs = []
-        for hit in results["hits"]:
-            for snippet in hit["snippets"]:
-                docs.append(Document(page_content=snippet))
-        return docs
+        if self.endpoint_type == "web":
+            results = requests.get(
+                f"https://api.ydc-index.io/search?query={query}",
+                headers=headers,
+            ).json()
+
+            docs = []
+            for hit in results["hits"]:
+                for snippet in hit["snippets"]:
+                    docs.append(Document(page_content=snippet))
+            return docs
+        else:
```

Comment:
slight preference to make this an explicit check for a value (eg. `elif self.endpoint_type == "snippet"`) and to raise an error if value is unknown